### DISCUSSION TITLE
Feature Suggestion (platform/bitbucket):  improve performance by adopting partial responses / sparse fieldsets

### DISCUSSION THREAD

**Discussion by setchy** [renovatebot/renovate#21379] [Suggest an Idea]
### Type of discussion.

I'm proposing an idea

### Tell us more.

Bitbucket Cloud APIs support [partial responses](https://developer.atlassian.com/cloud/bitbucket/rest/intro/#partial-response) / sparse fieldsets via the use of the `fields` query parameter.

By default, full resource representations are returned.  Using the `fields` query parameter allows you to select exactly which fields/nodes you want present in the API response object.

We use this approach heavily internally and depending on the API query and amount of data behind it, can lead to dramatic improvements in performance.

To illustrate, here are some metrics from one of our sandbox repos with 25 open PRs

| Response Type | Response Time | Page Size |
|------|------|-------|
| Full | 2.9 seconds | default page size (10 records) |
| Full | 4.17 seconds | 25 records |
| Partial | 280 milliseconds | 25 records |

I believe it should be possible to use the response object types to generate the partial fieldset selection, so that it reduces the effort of keeping the query parameters in sync with the expected response types.

**Comment by viceice** [comment]
- #20961

**Reply by setchy** [reply]
Ah, jinx. that's perfect!
